daemon/cmd/daemon_main.go

```diff
@@ -1307,6 +1311,18 @@ func initEnv(cmd *cobra.Command) {
 		}
 	}
 
+	// Setting node endpoint tunnel code here
+	log.Info("Setting the node tunnel address")
+	tunnelEndpointIPv4CIDR := option.Config.GetTunnelEndpointIPv4CIDR()
+	if tunnelEndpointIPv4CIDR != nil && option.Config.Tunnel != option.TunnelDisabled && option.Config.EnableIPv4 {
+		if _, intfIP, _, err := cidr.DetectCIDRByCIDR(tunnelEndpointIPv4CIDR, 4); err != nil {
+			log.WithField(logfields.CIDR, tunnelEndpointIPv4CIDR.IPNet.String()).Fatal("Can't detect the cidr for an actual interface IP")
+		} else {
+			log.Infof("Now set the node tunnel address: %s", intfIP.String())
```

Comment:
Two comments here:
* Please avoid Foof() variants on the log functions, instead when possible use `log.WithFields(...).Info(...)`.
* I don't think that this log is necessary at Info level as the user configuration is already printed during startup. Maybe this is good for debugging but in that case please switch to `log.Debug(...)`.